admin/server/auth/handlers.go

```diff
@@ -261,45 +302,107 @@ func (a *Authenticator) authLoginCallback(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	redirectURL, err := url.Parse(redirect)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to parse redirect URL: %s", err), http.StatusInternalServerError)
+		return
+	}
+
+	if a.admin.URLs.IsCustomDomain(redirectURL.Hostname()) {
```

Comment:
See comment above, not sure this will work.

If in the first call to `/auth/login` you notice it's a custom domain, consider storing that info in the cookie (e.g. `sess.Values[cookieFieldCustomDomainRedirect]`), so you don't need to do another lookup of that here. Might also make some of the URL handling simpler (no need to parse with `url.Parse`).